src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/DeleteUserProfileServiceImpl.java

```diff
@@ -30,16 +37,66 @@
     @Autowired
     private AuditService auditService;
 
+    @Autowired
+    private IdamFeignClient idamClient;
+
     @Value("${loggingComponentName}")
     private String loggingComponentName;
 
+    @Override
+    @Transactional
+    public UserProfilesDeletionResponse deleteByUserId(String userId) {
+        Response idamResponse = idamClient.deleteUser(userId);
+
+        if (idamResponse.status() == NO_CONTENT.value() || idamResponse.status() == NOT_FOUND.value()) {
+            UserProfile userProfile = validateUserStatus(userId);
+            return deleteUserProfiles(singletonList(userProfile));
+
+        } else {
+            UserProfilesDeletionResponse deletionResponse = new UserProfilesDeletionResponse();
+            deletionResponse.setMessage("IDAM Delete request failed for userId: " + userId);
+            deletionResponse.setStatusCode(idamResponse.status());
+            return deletionResponse;
+        }
+    }
+
+    @Override
+    @Transactional
+    public UserProfilesDeletionResponse deleteByEmailPattern(String emailPattern) {
+
+        List<UserProfile> userProfiles = userProfileRepository.findByEmailIgnoreCaseContaining(emailPattern);
+
+        if (userProfiles.isEmpty()) {
+            throw new ResourceNotFoundException("No User Profiles found for email pattern: " + emailPattern);
+        }
+
+        for (int i = userProfiles.size() - 1; i >= 0; i--) {
+            Response idamResponse = idamClient.deleteUser(userProfiles.get(i).getIdamId());
+
+            if (idamResponse.status() == NO_CONTENT.value() || idamResponse.status() == NOT_FOUND.value()) {
+                break;
```

Comment:
Correct me if I am wrong here. why are we breaking out of the loop if there is a success response?

Reply:
If the response is not successful then we do not want to delete them from UP so we remove them from the `userProfiles` list. This list is sent to the `deleteUserProfiles` method on line 89 for deletion in UP.

However, if the response is successful, then we do not have to remove them from the list and therefore we continue our loop.

Reply:
This has been addressed here: https://github.com/hmcts/rd-user-profile-api/pull/631/commits/da42b45bd8eb176b75fd12404b933ecd4a74f7af - thanks for your help @SujeethKumarAri 😄